src/pytest_cov/engine.py

```diff
@@ -124,12 +124,12 @@ def summary(self, stream):
         # Produce html report if wanted.
         if 'html' in self.cov_report:
             total = self.cov.html_report(ignore_errors=True, directory=self.cov_report['html'])
-            stream.write('Coverage HTML written to dir %s\n' % self.cov.config.html_dir)
+            stream.write('Coverage HTML written to dir %s\n' % self.cov_report['html'])
```

Comment:
this isn't quite right, because cov_report['html'] can be None

Reply:
https://github.com/pytest-dev/pytest-cov/pull/330/files#diff-3db14ea221c08dabe6d00f457c4cabbeR149-R156

Reply:
You make a good point (and the test suite needs to have a test added for it).  But I'm not sure the best solution. This is part of the problem with having a plugin provide the UI for coverage: it wants to report on what happened, but doesn't have access to the internal logic. 

Before I fixed the mutation of configuration, the plugin could examine the configuration after the fact to see where the report had been written.  Now the plugin has to duplicate some logic in order to accurately show where the HTML report was written.

Should Coverage.html_report have a verbose=True parameter so that it can write the message? Should it somehow provide the final configuration to the caller?

Reply:
I've pushed a new commit to deal with this.

Reply:
@nedbat I think the fix would be a function that takes a config object only

Reply:
@graingert sorry, i don't understand what that means.  A function where? That does what?

Reply:
I saw, I think I rebased on top of it correctly?